inputremapper/injection/mapping_handlers/combination_handler.py

```diff
@@ -109,74 +109,77 @@ def notify(
         is_pressed = event.value == 1
         self._pressed_keys[event.input_match_hash] = is_pressed
         # maybe this changes the activation status (triggered/not-triggered)
-        is_activated = self.is_activated()
+        changed = self.is_activated() != self._output_previously_active
+
+        if changed:
+            if is_pressed:
+                return self._handle_freshly_activated(suppress, event, source)
+            else:
+                return self._handle_freshly_deactivated(event, source)
+        else:
+            if is_pressed:
+                return self._handle_no_change_press(event)
+            else:
+                return self._handle_no_change_release(event)
```

Comment:
I like the idea, but I think this is a bit misleading. If I read this correctly `_handle_no_change_release` will be called when the state changes from suppressed to passive. I guess a simple comment should be fine.

Reply:
I'm a bit lost, could you please propose a change with the comment?